### ISSUE TITLE
`PLR6104` has false positive for non-commutative additions

### ISSUE THREAD

**Issue by charliermarsh** [astral-sh/ruff#10911]
@charliermarsh Is this fixed for:
```python
test_string = "desired = " + test_string  # False positive PLR6104
```
since `+` is sometimes commutative.

_Originally posted by @NeilGirdhar in https://github.com/astral-sh/ruff/issues/10900#issuecomment-2051918134_

**Comment by NeilGirdhar** [comment]
I'm not sure from the code, but you still catch:
```python
x = x + "string"  # should be x += "string"
```
right?

**Comment by NeilGirdhar** [comment]
Awesome, thanks for fixing this so fast! 😄

**Comment by ofek** [comment]
I ran into another case:

![image](https://github.com/astral-sh/ruff/assets/9677399/1f719766-dce5-495f-9672-01375573d0ba)

**Comment by charliermarsh** [comment]
This is fixed on `main` IIUC.
